kasparovsyncd/sync.go

```diff
@@ -235,238 +255,594 @@ func doesBlockExist(dbTx *gorm.DB, blockHash string) (bool, error) {
 	return !httpserverutils.IsDBRecordNotFoundError(dbErrors), nil
 }
 
-// addBlocks inserts all the data that could be gleaned out of the verbose
-// block and raw block data into the database. This includes transactions,
-// subnetworks, and addresses.
-// Note that if this function may take a nil dbTx, in which case it would start
-// a database transaction by itself and commit it before returning.
-func addBlock(client *jsonrpc.Client, rawBlock string, verboseBlock rpcmodel.GetBlockVerboseResult) error {
+func addBlocksAndTransactions(client *jsonrpc.Client, blocks []*rawAndVerboseBlock) error {
 	db, err := database.DB()
 	if err != nil {
 		return err
 	}
 	dbTx := db.Begin()
 	defer dbTx.RollbackUnlessCommitted()
 
-	// Skip this block if it already exists.
-	blockExists, err := doesBlockExist(dbTx, verboseBlock.Hash)
+	transactionIDtoTxWithMetaData, err := insertBlocksTransactions(dbTx, client, blocks)
```

Comment:
The fact that we have `insertBlocksTransactions` and `insertTransactionsBlocks` is quite confusing.
Please consider to rename somehow some of them.